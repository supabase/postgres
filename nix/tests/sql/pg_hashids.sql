select id_encode(1001); -- Result: jNl
select id_encode(1234567, 'This is my salt'); -- Result: Pdzxp
select id_encode(1234567, 'This is my salt', 10); -- Result: PlRPdzxpR7
select id_encode(1234567, 'This is my salt', 10, 'abcdefghijABCDxFGHIJ1234567890'); -- Result: 3GJ956J9B9
select id_decode('PlRPdzxpR7', 'This is my salt', 10); -- Result: 1234567
select id_decode('3GJ956J9B9', 'This is my salt', 10, 'abcdefghijABCDxFGHIJ1234567890'); -- Result: 1234567
