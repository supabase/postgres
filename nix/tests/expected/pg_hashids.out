select id_encode(1001); -- Result: jNl
 id_encode 
-----------
 jNl
(1 row)

select id_encode(1234567, 'This is my salt'); -- Result: Pdzxp
 id_encode 
-----------
 Pdzxp
(1 row)

select id_encode(1234567, 'This is my salt', 10); -- Result: PlRPdzxpR7
 id_encode  
------------
 PlRPdzxpR7
(1 row)

select id_encode(1234567, 'This is my salt', 10, 'abcdefghijABCDxFGHIJ1234567890'); -- Result: 3GJ956J9B9
 id_encode  
------------
 3GJ956J9B9
(1 row)

select id_decode('PlRPdzxpR7', 'This is my salt', 10); -- Result: 1234567
 id_decode 
-----------
 {1234567}
(1 row)

select id_decode('3GJ956J9B9', 'This is my salt', 10, 'abcdefghijABCDxFGHIJ1234567890'); -- Result: 1234567
 id_decode 
-----------
 {1234567}
(1 row)

